### PR TITLE
Add `embedderId` to `PointerEvent`

### PR DESCRIPTION
**Pull request by iskakaushik**
This field is used to look-up the event in the platform that resuted a given `PointerEvent`. This is currently only used on Android, where the `embedderId` is set to be the `motionEventId` for a given `MotionEvent`.

## Related Issues

https://github.com/flutter/flutter/issues/58837

## Tests

@blasten and I will work on testing strategy for this change as follow-up PRs.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

File: packages/flutter/lib/src/gestures/events.dart [hunk 1]
```diff
@@ -232,6 +233,8 @@ abstract class PointerEvent with Diagnosticable {
   }) : localPosition = localPosition ?? position,
        localDelta = localDelta ?? delta;
 
+  /// Corresponds to the motion event that resulted in this pointer event. (only android)
+  final int motionEventId;
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/events.dart:237]: Is this a unique ID for each `PointerEvent` or can multiple PointerEvents have the same motionEventId?

File: packages/flutter/lib/src/gestures/events.dart [hunk 2]
```diff
@@ -232,6 +233,8 @@ abstract class PointerEvent with Diagnosticable {
   }) : localPosition = localPosition ?? position,
        localDelta = localDelta ?? delta;
 
+  /// Corresponds to the motion event that resulted in this pointer event. (only android)
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/events.dart:236]: Can we document this in a more platform-independent way so we can re-use this for the other platforms as well?

Exact wording would depend on the answer to the question below. Does this id uniquely identify the event across framework and engine? Or (if there are multiple PointerEvent with this id) does it identify the native event that generated this framework-side PointerEvent uniquely on the engine side? Something like this should be included in the doc comment.

After that, we can also include a platform-specific paragraph that on android this maps to the ID of the underlying MotionEvent. Or something like that.

File: packages/flutter/lib/src/gestures/events.dart (hunk 1 above)
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/events.dart:237]: A name for this property that doesn't tie it to Android's MotionEvents (and hence can be reused for other platforms) would probably also be a good idea.

Comment (iskakaushik) [reply packages/flutter/lib/src/gestures/events.dart:237]: Nope, this is unique per `PointerEvent`. Multiple `PointerEvent`s can not have the same `motionEventId`.

File: packages/flutter/lib/src/gestures/events.dart (hunk 2 above)
Comment (iskakaushik) [reply packages/flutter/lib/src/gestures/events.dart:236]: I will try to address this and the comment below here: 

What do you think of `embedderPointerEventIdentifier` as a possible name for this? Given that this is different from `pointerIdentifier` -- which is used for hit testing, whereas `embedderPointerEventIdentifier` is used to tie back the current pointer event to the platform equivalent which resulted in this event.

If you agree with that, I can elaborate the dart-doc to explain the intent of this field along with the differences between this and `pointerIdentifier`.

Comment (goderbauer) [reply packages/flutter/lib/src/gestures/events.dart:236]: Can we just shorten this to `embedderId`?

Comment (goderbauer) [reply packages/flutter/lib/src/gestures/events.dart:236]: The doc should still include the explanation how it is different from the other IDs in this class.

Comment (iskakaushik) [reply packages/flutter/lib/src/gestures/events.dart:236]: SG

**Comment by iskakaushik** [comment]
@goderbauer I plan on landing this change as part of a manual engine roll during off-peak hours. The engine commit that this change relies on is at: https://github.com/flutter/engine/pull/19484.

File: packages/flutter/lib/src/gestures/events.dart [hunk 3]
```diff
@@ -232,6 +233,12 @@ abstract class PointerEvent with Diagnosticable {
   }) : localPosition = localPosition ?? position,
        localDelta = localDelta ?? delta;
 
+  /// Unique identifier that ties the [PointerEvent] to embedder event created it.
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/events.dart:236]: ... to the embedder event that created it?

File: packages/flutter/lib/src/gestures/events.dart [hunk 4]
```diff
@@ -232,6 +233,12 @@ abstract class PointerEvent with Diagnosticable {
...
+  ///
+  /// No two pointer events can have the same [embedderId]. This is different from
+  /// [pointer] identifier - used for hit-testing, whereas [embedderId] is used to
+  /// identify the platform event.
+  final int embedderId;
```
Comment (goderbauer) [inline packages/flutter/lib/src/gestures/events.dart:241]: Maybe add a section about Android explaining that on Android this is the ID of the underlying MotionEvent.

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by blasten — APPROVED** [review]
LGTM
